yt/frontends/stream/data_structures.py

```diff
@@ -890,7 +890,18 @@ def load_amr_grids(grid_data, domain_dimensions,
             cell_edges = np.linspace(domain_left_edge[iax],
                                      domain_right_edge[iax],
                                      dims[iax], endpoint=False)
-            if set(grid_left_edges[idx, iax]) - set(cell_edges):
+
+            # check that every value in cell_edges closely approximates
+            # a member of grid_left_edges[idx, iax]
+            # (account for rounding errors up to machine precision)
+            atol = grid_left_edges[idx, iax].min() / 1e3
+            found = np.zeros_like(grid_left_edges[idx, iax], dtype=bool)
+            for i, gle in enumerate(grid_left_edges[idx, iax]):
+                matches = np.isclose(cell_edges, gle, atol=atol, rtol=1e-16)
+                # check for uniqueness
+                found[i] = len(np.nonzero(matches)[0]) == 1
```

Comment:
Wouldn't it be fun if in a few years it turns out there was some weird edge case where this could be 2 or more?!  :rofl:

Reply:
Let's remove this line and find out ! 😇